integration_tests/sdk/setup_integration.py

```diff
@@ -0,0 +1,192 @@
+
+    # Find all the objects that already exist.
+    existing_names = set()
+    for object_name in demo_db_tables():
+        try:
+            s3.file(object_name, artifact_type=ArtifactType.TABLE, format="parquet")
+            existing_names.add(object_name)
+        except:
+            # Failing to fetch simply means we will need to populate this data.
+            pass
+
+    _add_missing_artifacts(client, s3, existing_names)
+
+
+def setup_data_integrations(filter_to: Optional[str] = None) -> None:
+    """Connects to the given integration name if the server hasn't yet. It also ensures
+    that the appropriate data is populated.
+
+    If `filter_to` is set, we only connect to that given integration name. Otherwise,
+    we attempt to connect to every integration listed in the config file.
+    """
+    if filter_to is not None:
+        data_integrations = [filter_to]
+    else:
+        data_integrations = list_data_integrations()
+
+    # No need to do any setup for the demo db.
+    if "aqueduct_demo" in data_integrations:
+        data_integrations.remove("aqueduct_demo")
+
+    if len(data_integrations) == 0:
+        return
+
+    test_config = _parse_config_file()
+    assert "data" in test_config
+
+    client = Client(*get_aqueduct_config())
+    for integration_name in data_integrations:
+        integration_config = test_config["data"][integration_name]
+        service_type = integration_config["type"]
+
+        # Connect to any integrations that don't exist.
+        connected_integrations = client.list_integrations()
+        if integration_name not in connected_integrations.keys():
+
+            # Modifying the config dictionary should be ok, since we only ever process
+            # an entry once.
+            del integration_config["type"]
+            client.connect_integration(integration_name, service_type, integration_config)
+
+        integration = client.integration(integration_name)
+
+        # Setup the data in each of these integrations.
+        if service_type == ServiceType.SNOWFLAKE:
+            _setup_snowflake_data(client, integration)
+        elif service_type == ServiceType.S3:
+            _setup_s3_data(client, integration)
+        else:
+            raise Exception("Test suite does not yet support %s." % service_type)
+
+
+def list_data_integrations() -> List[str]:
+    """Get the list of data integrations from the config file."""
+    test_config = _parse_config_file()
+    assert "data" in test_config
+    data_integrations = list(test_config["data"].keys())
+    data_integrations.insert(0, "aqueduct_demo")
+    return data_integrations
+
+
+def get_aqueduct_config() -> Tuple[str, str]:
+    # Returns the apikey and server address.
+    test_config = _parse_config_file()
+    assert (
+        "apikey" in test_config and "address" in test_config
+    ), "apikey and address must be set in test-config.yml."
+    return test_config["apikey"], test_config["address"]
```

Comment:
do you plan to use pydantic to parse this in the future?

Reply:
That's a good idea, made a task https://linear.app/aqueducthq/issue/ENG-2237/use-pydantic-for-test-config-struct-instead-of-dictionary